app/assets/javascripts/discourse/controllers/ignore-duration-with-username.js.es6

```diff
@@ -0,0 +1,31 @@
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import User from "discourse/models/user";
+
+export default Ember.Controller.extend(ModalFunctionality, {
+  loading: false,
+  ignoredUntil: null,
+  ignoredUsername: null,
+  actions: {
+    ignore() {
+      if (!this.get("ignoredUntil") || !this.get("ignoredUsername")) {
+        this.flash(
+          I18n.t("user.user_notifications.ignore_duration_time_frame_required"),
+          "alert-error"
+        );
+        return;
+      }
+      this.set("loading", true);
+      User.findByUsername(this.get("ignoredUsername")).then(user => {
+        user
+          .updateNotificationLevel("ignore", this.get("ignoredUntil"))
+          .then(() => {
+            this.get("onSuccess")(this.get("ignoredUsername"));
```

Comment:
Why isn't this sending an action instead?

Reply:
Communication from a controller to its route can be done using `send`, but here we are communicating from a controller to its calling component.

Reply:
We would / could have used `sendAction` but that's deprecated.